compiler/rustc_parse/src/parser/mod.rs

```diff
@@ -994,22 +994,22 @@ impl<'a> Parser<'a> {
 
     /// Advance the parser by one token.
     pub fn bump(&mut self) {
-        let (mut next, spacing) = self.token_cursor.inlined_next(self.desugar_doc_comments);
+        let mut next = self.token_cursor.inlined_next(self.desugar_doc_comments);
         self.token_cursor.num_next_calls += 1;
         // We've retrieved an token from the underlying
         // cursor, so we no longer need to worry about
         // an unglued token. See `break_and_eat` for more details
         self.token_cursor.break_last_token = false;
-        if next.span.is_dummy() {
+        if next.0.span.is_dummy() {
```

Comment:
It's awful that it has to be done for performance.
I pretty much never use `tuple.0` indexing and immediately destructure for readability.
I remember asking Aaron1011 to use destructuring on this specific line during review, when he originally added the spacing treatment to this code.

Reply:
I agree. I tried changing a couple of other hot functions (`inlined_next`, `inlined_bump_with`) that use the same kind of destructuring, but there was no performance change, so I left them alone.

Reply:
I'll add a comment explaining this.